### PR TITLE
aligning GitHub Actions workflows for build & release with other repos

### PR DESCRIPTION
**Pull request by dlockhart**
Yay, the first repo I've come across that already has some GHAs! 🎉 

Jeremy we chatted in Slack, but these changes align this repo's GHAs with what we've been applying to the others -- a separate workflow for "CI" vs. "release" and releasing via semantic-release. Let me know if you have any questions/concerns.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
-jobs:
-  build:
-    name: build
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: Brightspace/third-party-actions@actions/checkout
-      - name: Read configs
-        id: configs
-        run: echo "##[set-output name=NODE_VERSION;]$(cat .nvmrc)"
```
Comment (dlockhart) [inline .github/workflows/build.yml:18]: I don't _think_ the Node version in the NVM file is necessary... we'll see I guess.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,36 +0,0 @@
...
-          node-version: ${{ steps.configs.outputs.NODE_VERSION }}
-          registry-url: 'https://registry.npmjs.org'
-      - name: Install packages
-        run: npm i
-      - name: Run tests
-        if: (!startsWith(github.ref, 'refs/tags/')) # run on all but tags
-        run: npm run test
-        env:
-          SAUCE_USERNAME: lweber-d2l
-          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_LABS_ACCESS_KEY }}
```
Comment (dlockhart) [inline .github/workflows/build.yml:31]: I renamed this secret to align with the others and will remove this one once this merges.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      - uses: Brightspace/third-party-actions@actions/checkout
+      - uses: Brightspace/third-party-actions@actions/setup-node
+      - name: Install dependencies
+        run: npm install
+      - name: Lint
+        run: npm run lint
+      - name: Unit Tests (Headless)
+        run: npm run test:headless
+        env:
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_LABS_ACCESS_KEY }}
```
Comment (svanherk) [inline .github/workflows/ci.yml:17]: You mentioned renaming this in a comment above but looks to be the same as before?

Comment (dlockhart) [reply .github/workflows/ci.yml]: Just checking if you were paying attention! 😐

**Review by svanherk — APPROVED** [review]

**Review by JeremyWinkler — APPROVED** [review]

**Pull request merged**